### PR TITLE
Added A* algorithm

### PR DESCRIPTION
**Pull request by jeffin07**
### **Describe your change:**

Implemented A* algorithm 

* [ ] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [ ] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [ ] This pull request is all my own work -- I have not plagiarized.
* [ ] I know that pull requests will not be merged if they fail the automated tests.
* [ ] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [ ] All new Python files are placed inside an existing directory.
* [ ] All filenames are in all lowercase characters with no spaces or dashes.
* [ ] All functions and variable names follow Python naming conventions.
* [ ] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [ ] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by cclauss** [comment]
Please click the __Details__ links next to the ❌ below to see why our tests are failing.  This first one is spellchecker and the second one is a linter.

**Comment by cclauss** [comment]
ImportMismatchError?  https://travis-ci.com/github/TheAlgorithms/Python/builds/162981274#L800

**Comment by jeffin07** [comment]
@cclauss any idea about what causing 

> ImportMismatchError?

File: machine_learning/astar.py [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+
+def astar(world, start, goal):
+    '''
+    >>> p = Gridworld()
+    >>> start = Cell()
+    >>> start.position = (0,0)
+    >>> goal = Cell()
+    >>> goal.position = (4,4)
+    >>> astar(p, start, goal)
+
```
Comment (cclauss) [inline machine_learning/astar.py:102]:
```suggestion
    [(0, 0), (1, 1), (2, 2), (3, 3), (4, 4)]
    [(0, 0), (1, 1), (2, 2), (3, 3), (4, 4)]
```
The doctest requires the expected results.

File: machine_learning/a_star.py [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+				if c == n and c.f < n.f:
+					continue
+			_open.append(n)
+	path = []
+	while current.parent is not None:
+		path.append(current.position)
+		current = current.parent
+	path.append(current.position)
+	path = path[::-1]
+	print(path)
```
Comment (itsvinayak) [inline machine_learning/a_star.py]: delete this line, it is causing the build to fail 
`- print(path)`

**Comment by cclauss** [comment]
Why was this merged with failing tests?

**Pull request merged**